sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/security/CertificateNamesGenerator.java

```diff
@@ -0,0 +1,64 @@
+package com.mesosphere.sdk.offer.evaluate.security;
+
+import com.mesosphere.sdk.api.EndpointUtils;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.asn1.x509.GeneralNames;
+
+/**
+ * A {@link CertificateNamesGenerator} creates relevant names for given service pod.
+ */
+public class CertificateNamesGenerator {
+
+    private String serviceName;
+    private String taskName;
+
+    public CertificateNamesGenerator(String serviceName, String taskName) {
+        this.serviceName = serviceName;
+        this.taskName = taskName;
+    }
+
+    /**
+     * Returns a Subject for service certificate.
+     * @return
+     */
+    public X500Name getSubject() {
+       // TODO(mh): Should we somehow improve certificate name here? Right now it reflects only service
```

Comment:
So there is a single Common Name and then there are Subject Alternative Name values. I am going to add `autoip` name to the `CN` based on https://knowledge.rapidssl.com/support/ssl-certificate-support/index?page=content&actp=CROSSLINK&id=SO7239.

>The Common Name (CN), also known as the Fully Qualified Domain Name (FQDN), is the characteristic value within a Distinguished Name. Typically, it is composed of Host Domain Name and looks like, "www.symantec.com" or "symantec.com". The Common Name field is often misinterpreted and is filled out incorrectly. Do not use your organization's actual name as your common name.

Based on https://docs.mesosphere.com/1.9/networking/dns-overview/ it seems that `autoip` should support accessing service on and off virtual networks.

Example name: `kafka-0-broker.kafka.autoip.dcos.thisdcos.directory`